src/main/java/ratpack/zipkin/internal/ZipkinHttpClientImpl.java

```diff
@@ -28,107 +29,123 @@
-        requestInterceptor
-            .handle(requestAdapterFactory.createAdaptor(requestSpec.head(), HttpMethod.HEAD.getName()))
-    ));
-  }
-
-  @Override
-  public Promise<ReceivedResponse> options(final URI uri, final Action<? super RequestSpec> requestConfigurer) {
-    return request(uri, requestConfigurer.append(requestSpec ->
-        requestInterceptor
-            .handle(requestAdapterFactory.createAdaptor(requestSpec.options(), HttpMethod.OPTIONS.getName()))
-    ));
-  }
-
-  @Override
-  public Promise<StreamedResponse> requestStream(URI uri, HttpMethod method, final Action<? super RequestSpec> requestConfigurer) {
-
-    final Action<? super RequestSpec> wrappedRequestSpec = requestConfigurer.append(requestSpec ->
-      requestInterceptor
-        .handle(requestAdapterFactory.createAdaptor(requestSpec.method(method), method.getName()))
-    );
-
-    return delegate
-        .requestStream(uri, wrappedRequestSpec)
-        .wiretap(streamedResponseResult ->
-              responseInterceptor.handle(responseAdapterFactory.createAdapter(streamedResponseResult.getValue()))
+    private final HttpClient delegate;
+    private final ClientRequestInterceptor requestInterceptor;
+    private final ClientResponseInterceptor responseInterceptor;
+    private final ClientRequestAdapterFactory requestAdapterFactory;
+    private final ClientResponseAdapterFactory responseAdapterFactory;
+
+    @Inject
+    public ZipkinHttpClientImpl(final HttpClient delegate,
+                                final ClientRequestInterceptor requestInterceptor,
+                                final ClientResponseInterceptor responseInterceptor,
+                                final ClientRequestAdapterFactory requestAdapterFactory,
+                                final ClientResponseAdapterFactory responseAdapterFactory) {
+        this.delegate = delegate;
+        this.requestInterceptor = requestInterceptor;
+        this.responseInterceptor = responseInterceptor;
+        this.requestAdapterFactory = requestAdapterFactory;
+        this.responseAdapterFactory = responseAdapterFactory;
+    }
+
+    @Override
+    public ByteBufAllocator getByteBufAllocator() {
+        return delegate.getByteBufAllocator();
+    }
+
+    @Override
+    public int getPoolSize() {
+        return delegate.getPoolSize();
+    }
+
+    @Override
+    public Duration getReadTimeout() {
+        return delegate.getReadTimeout();
+    }
+
+    @Override
+    public int getMaxContentLength() {
+        return delegate.getMaxContentLength();
+    }
+
+    @Override
+    public void close() {
+        delegate.close();
+    }
+
+    @Override
+    public Promise<ReceivedResponse> request(URI uri, Action<? super RequestSpec> action) {
+        return delegate.request(uri, action);
+    }
+
+    @Override
+    public Promise<StreamedResponse> requestStream(URI uri, Action<? super RequestSpec> requestConfigurer) {
+        return delegate.requestStream(uri, requestConfigurer);
```

Comment:
I think this creates a similar situation in that a consumer of this ZipkinHttpClient is probably going to expect that using the methods in this class integrate with Zipkin, but in the case of `request` and `requestStream` no integration is provided.  

As far as I can tell, the only thing we need special for all this to work is the HTTP Method being invoked.  What do you think about somehow invoking `action` to grab that info?  Then we could even get rid of the extra `request` method added to the interface above and it is truly a drop-in replacement.